streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/core/master/resourcemanager/ResourceManager.java

```diff
@@ -0,0 +1,51 @@
+package org.ray.streaming.runtime.core.master.resourcemanager;
+
+import java.util.List;
+import java.util.Map;
+
+import org.ray.streaming.runtime.core.graph.executiongraph.ExecutionVertex;
+import org.ray.streaming.runtime.core.master.scheduler.strategy.SlotAssignStrategy;
+import org.ray.streaming.runtime.core.resource.Container;
+import org.ray.streaming.runtime.core.resource.Resources;
+
+/**
+ * The central role of resource management in JobMaster.
+ */
+public interface ResourceManager {
+
+  /**
+   * Get all registered container as a list.
+   *
+   * @return A list of containers.
+   */
+  List<Container> getRegisteredContainers();
+
+  /**
+   * Allocate resource to actor.
+   *
+   * @param exeVertex The specified worker vertex.
+   * @return Allocated resource.
+   */
+  Map<String, Double> allocateResource(final ExecutionVertex exeVertex);
```

Comment:
I think we can use more resource related parameters as methods parameters instead of `ExecutionVertex`. `ExecutionVertex ` should be coupled with `ResourceManager `. And it's difficult to know what information `ResourceManager ` need to allocate/deallocate resource.

Reply:
Already use container and request resource replace ```ExecutionVertex```.